src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -224,9 +240,7 @@ class DeepDiveLogParser extends JavaTokenParsers {
     | bcond
     )
   def bcond : Parser[Cond] =
-    ( expr ~ compareOperator ~ expr ^^ { case (lhs ~ op ~ rhs) =>
-        ComparisonCond(lhs, op, rhs)
-      }
+    ( cexpr ^^ ExprCond
```

Comment:
You mean such rule should be rejected from the parser?  I'd say those should be type errors rather than parse errors.  For example, one may want to write something like `R(x) :- S(x, is_true), is_true.` where `is_true` is obviously a boolean column/variable and an `lexpr`.  Trying to exclude certain type errors at the syntax level may be difficult and be too strict, so I'd rather add more rigorous type checking rather than degenerating the grammar.